### PR TITLE
refactor: Collect all status in one place

### PR DESCRIPTION
**Pull request by dashmage**
* Set all statuses from the collect_unit_status event handler.
* Change name from exporter_enabled to relation_present
* Refactor code into render_exporter_config
* Change return format for validate_exporter_configs to be simpler.

### CODE REVIEW

**Comment by dashmage** [comment]
A lot of unit tests break with this change. Once the overall logic is finalized, I'll go ahead and make the required unit test modifications.

**Comment by dashmage** [comment]
I'm thinking whether I should make changes to resolve https://github.com/canonical/hardware-observer-operator/issues/196 as well since I'm anyway changing the stored state variables with this PR.

File: src/charm.py [hunk 1]
```diff
@@ -74,88 +78,84 @@ def __init__(self, *args: Any) -> None:
 
         self._stored.set_default(
             config={},
-            exporter_installed=False,
-            resource_installed=False,
```
Comment (chanchiwai-ray) [inline src/charm.py:78]: It's better that we avoid using store for `*_installed` here, and create methods to directly verify them

**Comment by Pjack** [comment]
Close this PR, since it was pending for a while.

**Pull request closed without merging**